pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -0,0 +1,150 @@
+		})
+
+		When("the pod link is defined", func() {
+			BeforeEach(func() {
+				podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: 1000}}
+				handler.EXPECT().LinkByName(ifaceName).Return(podLink, nil)
+			})
+
+			When("the pod's IP addresses are defined", func() {
+				BeforeEach(func() {
+					podIP = netlink.Addr{IPNet: &net.IPNet{IP: net.IPv4(10, 35, 0, 6), Mask: net.CIDRMask(24, 32)}}
+					handler.EXPECT().AddrList(podLink, netlink.FAMILY_V4).Return([]netlink.Addr{podIP}, nil)
+				})
+
+				When("the pod's routes are defined", func() {
+					BeforeEach(func() {
+						defaultGwRoute = netlink.Route{Dst: nil, Gw: net.IPv4(10, 35, 0, 1)}
+						handler.EXPECT().RouteList(podLink, netlink.FAMILY_V4).Return([]netlink.Route{defaultGwRoute}, nil)
+					})
+
+					It("should succeed reading the pod link info", func() {
+						Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+						Expect(bridgeConfigurator.podNicLink).To(Equal(podLink))
+					})
+
+					It("should succeed reading the pod IP addresses", func() {
+						Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+						Expect(bridgeConfigurator.podIfaceIP).To(Equal(podIP))
+					})
+
+					It("should succeed reading the routes", func() {
+						Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+						Expect(bridgeConfigurator.podIfaceRoutes).To(ConsistOf(defaultGwRoute))
+					})
+				})
+
+				When("the pod does not feature routes", func() {
+					BeforeEach(func() {
+						handler.EXPECT().RouteList(podLink, netlink.FAMILY_V4).Return([]netlink.Route{}, nil)
+					})
+
+					It("should fail", func() {
+						Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(HaveOccurred())
+					})
+				})
+			})
+
+			When("the pod does not report IP addresses", func() {
+				BeforeEach(func() {
+					handler.EXPECT().AddrList(podLink, netlink.FAMILY_V4).Return([]netlink.Addr{}, nil)
+				})
+
+				It("should report disabled IPAM and miss the IP address field", func() {
+					Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+					Expect(bridgeConfigurator.podIfaceIP).To(Equal(netlink.Addr{}))
+					Expect(bridgeConfigurator.ipamEnabled).To(BeFalse())
+				})
+
+				It("should not care about missing routes when an IP was not found", func() {
+					handler.EXPECT().RouteList(podLink, netlink.FAMILY_V4).Return([]netlink.Route{}, nil).Times(0)
+					Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+				})
+			})
+
+			When("the pod errors while attempting to retrieve the IP address", func() {
+				BeforeEach(func() {
+					podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: 1000}}
+					handler.EXPECT().AddrList(podLink, netlink.FAMILY_V4).Return([]netlink.Addr{}, fmt.Errorf("failed to read IPs"))
```

Comment:
Can we change the error string to denote that we are forcing it from a test so we don't think it's from production code ?

something like "bridge_test: forcing failure at read IPs".

Reply:
I fail to see the advantage; if you grep, you'll find it. 

I don't mind complying though

Reply:
The idea is that if you run the unit tests and you see a massive number of errors and one of them is "bridge_test: force..." you know for sure that is not at the production code.